### PR TITLE
Fixes in node builders

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Leftovers of PR #21637

### Tickets:
 - CVS-128451

### CODE REVIEW

File: src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ ov::NodeVector split_to_QKV(const std::shared_ptr<v1::Add>& node,
         // head_size = hidden_size / num_heads
         head_size = std::make_shared<v1::Divide>(hidden_size, num_heads_node);
         // split the node into 3 even parts Q, K, V with shape (batch_size, sequence_len, hidden_size)
-        split = ov::op::util::split(node, 3, 2);
+        split = ov::op::util::make_split(node, 3, 2);
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: as I see, the code from test utils is still used in the production code. Can we avoid it?

Comment (olpipi) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: @ilya-lavrenov @vurusovs 
This is not a test utils. This is a separated cmake target. But it is located in src/tests/...
I can move this target from src/tests/ov_helpers/ov_models/ov_builders to dev API.
Do you have objections to this approach?

Comment (vurusovs) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: As we discussed, there should be no CMake target for production builders, it should be fully moved to ONNX FE (if duplications required, let's duplicate it), and no includes from tests in production code should be presented

Comment (olpipi) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: They are unfortunately used not only in ONNX frontend. So we cannot do it

Comment (vurusovs) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: Let's make copy of them, the list should be very short

Comment (olpipi) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: Yes, it may take a place

Comment (olpipi) [reply src/frontends/onnx/frontend/src/op/com.microsoft/attention.cpp:156]: done. please re-review

File: src/frontends/onnx/frontend/src/op/softmax.cpp [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include "openvino/op/reshape.hpp"
 #include "openvino/op/shape_of.hpp"
 #include "openvino/op/softmax.hpp"
-#include "ov_models/ov_builders/reshape.hpp"
+#include "utils/ov_builders/reshape.hpp"
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/op/softmax.cpp:12]: @gkrivor do you think we need `ov_builders` in the middle or we can place directly to `utils`? Looks like `utils` already contain so-called builders

**Review by gkrivor — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**